inference-test-tool/test_inference_boxes.py

```diff
@@ -29,10 +29,15 @@ def generate_images_with_boxes(images, boxes, output_folder):
             ul = box['top_left']
             br = box['bottom_right']
             points = [tuple(ul), (br[0], ul[1]), tuple(br), (ul[0], br[1]), tuple(ul)]
-            draw.line(points, fill="red", width=5)
-            
+            draw.line(points, fill='red', width=5)
+            label = box['label']
+            font = ImageFont.truetype('OpenSans-Regular.ttf', br[1] // 20)
+            text_width, text_height = draw.textsize(label, font)
+            text_position = ((br[0] - text_width) / 2, (br[1] - text_height) / 2)
```

Comment:
Same happens to this calculation. It does not work for bounding boxes that do not span the whole image.